From 8f7f52da38dc95175f5b2f2dc937f1b29fe69788 Mon Sep 17 00:00:00 2001
From: Hari Subramaniyan <harisankar@apache.org>
Date: Thu, 18 Aug 2016 11:21:37 -0700
Subject: [PATCH 1016/1164] CDH-48309 HIVE-14552: TestParseNegative fix (Hari
 Subramaniyan, reviewed by Ashutosh Chauhan)

(cherry picked from commit 6d3085b7e5b686a7dc7d37df754816d34247c5e6)

Change-Id: Id0ef0fc45ce7f5c57c9a854a1e3247257c07ee4f
---
 .../hadoop/hive/ql/parse/CoreParseNegative.java    |   23 +++++++++-----------
 1 file changed, 10 insertions(+), 13 deletions(-)

diff --git a/itests/util/src/main/java/org/apache/hadoop/hive/ql/parse/CoreParseNegative.java b/itests/util/src/main/java/org/apache/hadoop/hive/ql/parse/CoreParseNegative.java
index 47d789a..4c62ed0 100644
--- a/itests/util/src/main/java/org/apache/hadoop/hive/ql/parse/CoreParseNegative.java
+++ b/itests/util/src/main/java/org/apache/hadoop/hive/ql/parse/CoreParseNegative.java
@@ -36,7 +36,7 @@
   private static QTestUtil qt;
 
   static CliConfigs.ParseNegativeConfig cliConfig = new CliConfigs.ParseNegativeConfig();
-
+  static boolean firstRun;
   public CoreParseNegative(AbstractCliConfig testCliConfig) {
     super(testCliConfig);
   }
@@ -47,7 +47,7 @@ public void beforeClass() {
     MiniClusterType miniMR = cliConfig.getClusterType();
     String initScript = cliConfig.getInitScript();
     String cleanupScript = cliConfig.getCleanupScript();
-
+    firstRun = true;
     try {
       String hadoopVer = cliConfig.getHadoopVersion();
       qt = new QTestUtil((cliConfig.getResultsDir()), (cliConfig.getLogDir()), miniMR, null,
@@ -67,26 +67,21 @@ public void setUp() {
   @Override
   @After
   public void tearDown() {
-    try {
-      qt.clearPostTestEffects();
-    } catch (Exception e) {
-      System.err.println("Exception: " + e.getMessage());
-      e.printStackTrace();
-      System.err.flush();
-      fail("Unexpected exception in tearDown");
-    }
   }
 
   @Override
   @AfterClass
   public void shutdown() throws Exception {
+    String reason = "clear post test effects";
     try {
+      qt.clearPostTestEffects();
+      reason = "shutdown";
       qt.shutdown();
     } catch (Exception e) {
       System.err.println("Exception: " + e.getMessage());
       e.printStackTrace();
       System.err.flush();
-      throw new RuntimeException("Unexpected exception in shutdown",e);
+      throw new RuntimeException("Unexpected exception in " + reason,e);
     }
   }
 
@@ -101,8 +96,10 @@ public void runTest(String tname, String fname, String fpath) throws Exception {
       System.err.println("Begin query: " + fname);
 
       qt.addFile(fpath);
-
-      qt.init(fname);
+      if (firstRun) {
+        qt.init(fname);
+        firstRun = false;
+      }
       ASTNode tree = qt.parseQuery(fname);
       List<Task<? extends Serializable>> tasks = qt.analyzeAST(tree);
       fail("Unexpected success for query: " + fname + debugHint);
-- 
1.7.9.5

